src/cpptools.ts

```diff
@@ -304,6 +309,8 @@ export class CppConfigurationProvider implements cpt.CustomConfigurationProvider
       compilerArgs: flags || undefined
     };
 
+    this._workspaceBrowseConfigurations.set(util.platformNormalizePath(opts.folder), this._workspaceBrowseConfiguration);
```

Comment:
This seems incorrect.  `this._workspaceBrowseConfiguration` is supposed to be the accumulation of all browse paths across all folders (it is returned by `provideBrowseConfiguration`).  I think you will need to create a local `cpt.WorkspaceBrowseConfiguration` in `updateConfigurationData` (instead of reinitializing this one) and pass it through to the functions it calls.

Reply:
this._workspaceBrowseConfiguration isn't actually be user-hittable by our C/C++ extension. Are you referring to possible other callers?

Reply:
I think the current behavior is okay unless we get complaints from consumers. The usage of the API is expected to be how our C/C++ handles it via check the per-workspace folder data first and ignore the other data.